s3/pkg/helper/string.go

```diff
@@ -0,0 +1,58 @@
+package helper
+
+import "strings"
+
+func HasBucketInDomain(host string, prefix string, domains []string) (ok bool, bucket string)  {
+	for _, d := range domains {
+		if strings.HasSuffix(host, prefix+d) {
+			return true, strings.TrimSuffix(host, prefix+d)
+		}
+	}
+	return false, ""
+}
+
+func StringInSlice(s string, ss []string) bool {
```

Comment:
please rename to declare intent of _contains_ or _find_
Typically, _contains_ will return a _bool_, and _find_ will return the _index_ where it was found (if it was found)